### DISCUSSION TITLE
有关p2p问题

### DISCUSSION THREAD

**Discussion by novada0** [SUSTech-CS305-Fall22/CS305-Project-Skeleton#38] [Q&A]
请问会出现peerA既作为peerB的发送方，又作为peerB的接受方的情况嘛？谢谢！

**Comment by TANSixu** [comment]
Hi, 这是有可能的，也是可以优化的地方。简单的实现是将其视为两个连接，各自有自己的cwnd，并且一个连接下载完后就丢掉，不再维护。可以优化的地方是
1. 融合两个连接，实现双工传输。
2. 保留连接信息，此后再用到就有现成的RTT和拥塞信息。

当然，这里的优化可能无法带来bonus test中性能的显著提升。如果实现了这部分内容，可以在presentation时展示，会获得相应分数的！
